### PR TITLE
Doc schema, filter, Date

### PR DESCRIPTION
**Pull request by MicLi01**
Leave feedback. I will move files to correct directory when the database connection items are merged.

### CODE REVIEW

File: server/checkdoc.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    let currentdate = Doc.gendate();
+    let docinst = new Doc({
+        name: req.body.name,
+        file: req.body.text,
+        owner_id: req.body.owner_id,
+        locked: 0,
+        revision_id: 0,
+        last_revised: currentDate
+    });
+};
```
Comment (chc5) [inline server/checkdoc.js]: Please use Google's JavaScript style guide when writing your code.
https://google.github.io/styleguide/jsguide.html

File: server/doc.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
+const mongoose = require('mongoose');
+mongoose.promise = Promise;
+const Schema = mongoose.Schema;
+const date = require('date-and-time');
+
+const docSchema = new Schema({
+    name: {type: String, required = true},
+    file: {type: String, required = true},
+    owner_id: {type: int, required = true},
+    locked: int,
```
Comment (melvinheks) [inline server/doc.js]: Why "int" over boolean?

File: server/doc.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
+const mongoose = require('mongoose');
+mongoose.promise = Promise;
+const Schema = mongoose.Schema;
+const date = require('date-and-time');
+
+const docSchema = new Schema({
+    name: {type: String, required = true},
+    file: {type: String, required = true},
+    owner_id: {type: int, required = true},
```
Comment (melvinheks) [inline server/doc.js]: int is not a valid type

File: server/doc.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    locked: int,
+    revision_id: int,
+    last_revised: {type: Date, required = true}
+})
+
+docSchema.methods.filter = function(file_text, taboo_words){
```
Comment (melvinheks) [inline server/doc.js]: This method does not change any of the values within a doc, i.e. this method does nothing.

Comment (melvinheks) [inline server/doc.js]: file_text should not be a parameter. This is a method on a document, it should be accessed with this.file

File: server/doc.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    //A String is considered a word if there is a space
+    //Checks all the words and see if any are in taboo words and replaces those words
+    let wordsarr = file_text.split(" ");
```
Comment (melvinheks) [inline server/doc.js]: If you're going to go with this approach, you have to join the array back afterwards and reassign it to this.file

File: server/doc.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    for(i = 0; i < wordsarr.length; i++){
```
Comment (melvinheks) [inline server/doc.js]: let i = 0, other option is to use iterator. for(let i in wordsarr){console.log(wordsarr[i])}

File: server/doc.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+        if(taboo_words.indexOf(wordsarr[i]) >= -1){
+            wordsarr[i] = "taboo";
+        }     
+    }
+};
+
+docSchema.methods.gendate = function(){
```
Comment (melvinheks) [inline server/doc.js]: Use a pre save hook instead of creating this method. https://mongoosejs.com/docs/middleware.html#pre

File: server/checkdoc.js (hunk 1 above)
Comment (melvinheks) [reply server/checkdoc.js]: +1

Comment (melvinheks) [inline server/checkdoc.js]: What is the purpose of this file?

File: server/doc.js [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+    revision_id: int,
+    last_revised: {type: Date, required = true}
+})
+
+docSchema.methods.filter = function(file_text, taboo_words){
+    //A String is considered a word if there is a space
+    //Checks all the words and see if any are in taboo words and replaces those words
+    let wordsarr = file_text.split(" ");
+    for(i = 0; i < wordsarr.length; i++){
+        if(taboo_words.indexOf(wordsarr[i]) >= -1){
```
Comment (melvinheks) [inline server/doc.js]: This does not account for punctuation or any other variation

File: server/doc.js (hunk 2 above)
Comment (MicLi01) [reply server/doc.js]: Changed to Boolean

File: server/doc.js (hunk 3 above)
Comment (MicLi01) [reply server/doc.js]: Changed to Number

File: server/doc.js (hunk 4 above)
Comment (MicLi01) [reply server/doc.js]: retrieved the attribute using this to modify it

Comment (MicLi01) [reply server/doc.js]: changed using this.file

File: server/doc.js (hunk 5 above)
Comment (MicLi01) [reply server/doc.js]: Joined the split and changed array back together.

File: server/doc.js (hunk 6 above)
Comment (MicLi01) [reply server/doc.js]: Added let to restrict scope

File: server/doc.js (hunk 7 above)
Comment (MicLi01) [reply server/doc.js]: Used pre save hook for adding the Date.

File: server/checkdoc.js (hunk 1 above)
Comment (MicLi01) [reply server/checkdoc.js]: added functionality

File: server/doc.js (hunk 8 above)
Comment (MicLi01) [reply server/doc.js]: handles separate words can create new method to handle all substrings

File: server/checkdoc.js (hunk 1 above)
Comment (melvinheks) [reply server/checkdoc.js]: What?

File: server/checkdoc.js [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+const inputdoc = (req, res) => {
+  let docinst = new Doc({
+    name: req.body.name,
+    file: req.body.text,
+    owner_id: req.body.owner_id,
+    locked: false,
+    revision_id: 0,
+    last_revised: null
+  });
+  let tabooSchema = new Schema({
```
Comment (melvinheks) [inline server/checkdoc.js]: This schema should not be defined here

File: server/checkdoc.js [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+    word: String
+  })
+  taboo = mongoose.model('Taboo', tabooSchema);
+  try {
+    const results = await taboo.find({});
```
Comment (melvinheks) [inline server/checkdoc.js]: Do not use await while the rest of the code uses call backs

File: server/doc.js [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+mongoose.promise = Promise;
+const Schema = mongoose.Schema;
+const date = require('date-and-time');
+
+const docSchema = new Schema({
+  name: {type: String, required = true},
+  file: {type: String, required = true},
+  owner_id: {type: Number, required = true},
+  locked: Boolean,
+  revision_id: Number,
```
Comment (melvinheks) [inline server/doc.js]: This should be a list of foreign keys i.e 
revisions:[
      {type: Schema.Types.ObjectId, ref: 'Revisions'}
]

File: server/doc.js [hunk 12]
```diff
@@ -0,0 +1,37 @@
+const mongoose = require('mongoose');
+mongoose.promise = Promise;
+const Schema = mongoose.Schema;
+const date = require('date-and-time');
+
+const docSchema = new Schema({
+  name: {type: String, required = true},
```
Comment (melvinheks) [inline server/doc.js]: Rename to title for clarity

File: server/doc.js [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+  file: {type: String, required = true},
```
Comment (melvinheks) [inline server/doc.js]: rename to content?

File: server/doc.js [hunk 14]
```diff
@@ -0,0 +1,37 @@
...
+  owner_id: {type: Number, required = true},
```
Comment (melvinheks) [inline server/doc.js]: Should be a foreign key i.e.
{type: Schema.Types.ObjectId, ref: 'User'}

File: server/doc.js [hunk 15]
```diff
@@ -0,0 +1,37 @@
...
+  locked: Boolean,
+  revision_id: Number,
+  last_revised: {type: Date, required = true}
+})
+
+docSchema.methods.filter = function(taboo_words){
```
Comment (melvinheks) [inline server/doc.js]: This method is not needed. We instead need to generate a list of indexes indicating the locations of all taboo words.

File: server/doc.js [hunk 16]
```diff
@@ -0,0 +1,37 @@
...
+  for(let i = 0; i < wordsarr.length; i++){
+      if(taboo_words.indexOf(wordsarr[i]) >= -1){
+          wordsarr[i] = "taboo";
+      }     
+  }
+  this.file = wordsarr.join(" ");
+};
+
+//docSchema.methods.gendate = function(){
+docSchema.pre('save', function(){
```
Comment (melvinheks) [inline server/doc.js]: This method is not needed. Use default approach as stated in a previous comment.

File: server/doc.js (hunk 12 above)
Comment (melvinheks) [inline server/doc.js]: required:true, not required=true
Fix for all

File: server/doc.js (hunk 13 above)
Comment (MicLi01) [reply server/doc.js]: renamed

File: server/doc.js (hunk 12 above)
Comment (MicLi01) [reply server/doc.js]: document name changed to title

File: server/checkdoc.js [hunk 17]
```diff
@@ -0,0 +1,31 @@
...
+    console.log(results);
+  } catch (err) {
+    throw err;
+  }
+  taboo_words_arr = results;
+  //Modify file
+  docinst.filter(taboo_words_arr);
```
[the remark this answers is not in the document]
Comment (MicLi01) [reply server/checkdoc.js]: filter returns array of tuples. 0 idx of a tuple is start and 1 idx is end + 1.

File: server/checkdoc.js (hunk 10 above)
Comment (MicLi01) [reply server/checkdoc.js]: changed to using callback

File: server/checkdoc.js (hunk 9 above)
Comment (MicLi01) [reply server/checkdoc.js]: separated into a different file

File: server/doc.js [hunk 18]
```diff
@@ -0,0 +1,37 @@
...
+const Schema = mongoose.Schema;
+const date = require('date-and-time');
+
+const docSchema = new Schema({
+  name: {type: String, required = true},
+  file: {type: String, required = true},
+  owner_id: {type: Number, required = true},
+  locked: Boolean,
+  revision_id: Number,
+  last_revised: {type: Date, required = true}
```
[the remark this answers is not in the document]
Comment (MicLi01) [reply server/doc.js]: removed method and used default for dates

File: server/doc.js (hunk 11 above)
Comment (MicLi01) [reply server/doc.js]: changed

File: server/doc.js (hunk 14 above)
Comment (MicLi01) [reply server/doc.js]: added in foreign key

File: server/doc.js (hunk 15 above)
Comment (MicLi01) [reply server/doc.js]: changed to return a arr of starts and ends for all places where taboo words show up

File: server/doc.js (hunk 16 above)
Comment (MicLi01) [reply server/doc.js]: uses Date.now now

File: server/doc.js (hunk 12 above)
Comment (MicLi01) [reply server/doc.js]: fixed syntax

File: server/checkdoc.js [hunk 19]
```diff
@@ -0,0 +1,32 @@
...
+const Taboo = require('taboo');
+
+const inputdoc = (req, res) => {
+  let docinst = new Doc({
+    title: req.body.name,
+    content: req.body.text,
+    owner_id: req.body.owner_id,
+    locked: false,
+    revision_id: 0,
+    last_revised: null
```
Comment (melvinheks) [inline server/checkdoc.js]: last_revised not needed

File: server/checkdoc.js [hunk 20]
```diff
@@ -0,0 +1,32 @@
...
+  });
+
+  Taboo.find({}, function (err, results) {
```
Comment (melvinheks) [inline server/checkdoc.js]: Why is this being called here?

File: server/checkdoc.js [hunk 21]
```diff
@@ -0,0 +1,32 @@
...
+    if (err) {
+      throw err;
+    }
+    else {
+      console.log(results);
+      var taboo_words_arr = results;
```
Comment (melvinheks) [inline server/checkdoc.js]: This will not work how you expect it to since this is a callback.
Also, do not use var.

File: server/checkdoc.js [hunk 22]
```diff
@@ -0,0 +1,32 @@
...
+    }
+  });
+
+  //Modify file
```
Comment (melvinheks) [inline server/checkdoc.js]: This does not modify the file

File: server/checkdoc.js [hunk 23]
```diff
@@ -0,0 +1,32 @@
...
+  docinst.filter(taboo_words_arr);
```
Comment (melvinheks) [inline server/checkdoc.js]: This should not be called here

File: server/checkdoc.js (hunk 21 above)
Comment (melvinheks) [inline server/checkdoc.js]: results is not an array of strings, it is an array of objects of the format defined in taboowords schema

File: server/doc.js [hunk 24]
```diff
@@ -0,0 +1,36 @@
...
+
+const docSchema = new Schema({
+  title: {type: String, required:true},
+  content: {type: String, required:true},
+  owner_id: {type: Schema.Types.ObjectId, ref: 'User'},
+  locked: Boolean,
+  revisions:[
+    {type: Schema.Types.ObjectId, ref: 'Revision'}
+  ],
+  last_revised: {type: Date,
```
Comment (melvinheks) [inline server/doc.js]: last_revised -> date_created

File: server/doc.js [hunk 25]
```diff
@@ -0,0 +1,36 @@
...
+    default: Data.now
+  }
+})
+
+docSchema.methods.filter = function(taboo_words){
+  //A String is considered a word if there is a space
+  //Checks all the words and see if any are in taboo words and replaces those words
+  //returns indexes of all taboo words, format at each index: [start, end + 1]
+  idx_list = [];
+  for(let i = 0; i < taboo_words.length; i++){
```
Comment (melvinheks) [inline server/doc.js]: since you are not assigning values to the indexes of taboo_words, use an iterative for loop, i.e 
for(const taboo of taboo_words){
}

File: server/doc.js [hunk 26]
```diff
@@ -0,0 +1,36 @@
...
+    let current_idx = 0;
+    let start_idx = 0;
+    while((start_idx = this.content.indexOf(taboo_words[i], current_idx)) > -1){
+      idx_list[idx_list] = [start_idx, start_idx + len(taboo_words[i])];
```
Comment (melvinheks) [inline server/doc.js]: idx_list[idx_list] ??

File: server/taboo.js [hunk 27]
```diff
@@ -0,0 +1,14 @@
+const mongoose = require('mongoose');
+mongoose.promise = Promise;
+const Schema = mongoose.Schema;
+const date = require('date-and-time');
+
+const tabooWordsSchema = new Schema({
+  word: {type: String, required: true},
+  suggesterId: {type: Number, required: true},
```
Comment (melvinheks) [inline server/taboo.js]: this should be a foreign key to user

File: server/taboo.js [hunk 28]
```diff
@@ -0,0 +1,14 @@
...
+  date: {type: Date, default: Date.now},
+  approverId: Number
```
Comment (melvinheks) [inline server/taboo.js]: this should be a foreign key to user

File: server/doc.js [hunk 29]
```diff
@@ -0,0 +1,36 @@
+const mongoose = require('mongoose');
+mongoose.promise = Promise;
+const Schema = mongoose.Schema;
+const date = require('date-and-time');
+
+const docSchema = new Schema({
```
Comment (melvinheks) [inline server/doc.js]: docSchema -> DocSchema

File: server/taboo.js [hunk 30]
```diff
@@ -0,0 +1,14 @@
+const mongoose = require('mongoose');
+mongoose.promise = Promise;
+const Schema = mongoose.Schema;
+const date = require('date-and-time');
+
+const tabooWordsSchema = new Schema({
```
Comment (melvinheks) [inline server/taboo.js]: tabooWordsSchema -> TabooWordsSchema

File: server/doc.js [hunk 31]
```diff
@@ -0,0 +1,36 @@
...
+  locked: Boolean,
+  revisions:[
+    {type: Schema.Types.ObjectId, ref: 'Revision'}
+  ],
+  last_revised: {type: Date, 
+    default: Data.now
+  }
+})
+
+docSchema.methods.filter = function(taboo_words){
```
Comment (melvinheks) [inline server/doc.js]: Rename

File: server/checkdoc.js (hunk 19 above)
Comment (MicLi01) [reply server/checkdoc.js]: removed last_revised

File: server/checkdoc.js (hunk 21 above)
Comment (MicLi01) [reply server/checkdoc.js]: rearranged variable

File: server/checkdoc.js (hunk 22 above)
Comment (MicLi01) [reply server/checkdoc.js]: removed

File: server/checkdoc.js (hunk 23 above)
Comment (MicLi01) [reply server/checkdoc.js]: removed

File: server/checkdoc.js (hunk 21 above)
Comment (MicLi01) [reply server/checkdoc.js]: changed name

File: server/doc.js (hunk 24 above)
Comment (MicLi01) [reply server/doc.js]: changed name

File: server/doc.js (hunk 26 above)
Comment (MicLi01) [reply server/doc.js]: changed index to length

File: server/taboo.js (hunk 27 above)
Comment (MicLi01) [reply server/taboo.js]: added in foreign key

File: server/taboo.js (hunk 28 above)
Comment (MicLi01) [reply server/taboo.js]: added in foreign key

File: server/doc.js (hunk 29 above)
Comment (MicLi01) [reply server/doc.js]: changed name

File: server/taboo.js (hunk 30 above)
Comment (MicLi01) [reply server/taboo.js]: changed name

File: server/doc.js (hunk 31 above)
Comment (MicLi01) [reply server/doc.js]: changed name

File: server/checkdoc.js (hunk 21 above)
Comment (melvinheks) [reply server/checkdoc.js]: ?

File: server/taboo.js [hunk 32]
```diff
@@ -0,0 +1,14 @@
+const mongoose = require('mongoose');
+mongoose.promise = Promise;
+const Schema = mongoose.Schema;
+const date = require('date-and-time');
+
+const TabooWordsSchema = new Schema({
+  word: {type: String, required: true},
+  suggesterId: {type: Number, ref: 'User'},
```
Comment (melvinheks) [inline server/taboo.js]: The type for a foreign key should be type: Schema.Types.ObjectId

File: server/taboo.js [hunk 33]
```diff
@@ -0,0 +1,14 @@
...
+  date: {type: Date, default: Date.now},
+  approverId: {type: Number, ref: 'User'}
```
Comment (melvinheks) [inline server/taboo.js]: type: Schema.Types.ObjectId

File: server/taboo.js [hunk 34]
```diff
@@ -0,0 +1,14 @@
+const mongoose = require('mongoose');
+mongoose.promise = Promise;
+const Schema = mongoose.Schema;
+const date = require('date-and-time');
```
Comment (melvinheks) [inline server/taboo.js]: Why is this needed?

File: server/doc.js [hunk 35]
```diff
@@ -0,0 +1,36 @@
...
+
+const DocSchema = new Schema({
+  title: {type: String, required:true},
+  content: {type: String, required:true},
+  owner_id: {type: Schema.Types.ObjectId, ref: 'User'},
+  locked: Boolean,
+  revisions:[
+    {type: Schema.Types.ObjectId, ref: 'Revision'}
+  ],
+  date_created: {type: Date,
```
Comment (melvinheks) [inline server/doc.js]: type should be moved to the next line for proper formatting, or just move default up

File: server/doc.js (hunk 25 above)
Comment (melvinheks) [reply server/doc.js]: Please address this

File: server/doc.js [hunk 36]
```diff
@@ -0,0 +1,36 @@
...
+DocSchema.methods.findAllTabooIdx = function(taboo_words){
+  //A String is considered a word if there is a space
+  //Checks all the words and see if any are in taboo words and replaces those words
+  //returns indexes of all taboo words, format at each index: [start, end + 1]
+  idx_list = [];
+  for(let i = 0; i < taboo_words.length; i++){
+    let current_idx = 0;
+    let start_idx = 0;
+    while((start_idx = this.content.indexOf(taboo_words[i], current_idx)) > -1){
+      idx_list[length] = [start_idx, start_idx + len(taboo_words[i])];
```
Comment (melvinheks) [inline server/doc.js]: length is not a variable? What are you trying to accomplish?

File: server/checkdoc.js [hunk 37]
```diff
@@ -0,0 +1,28 @@
+const Doc = require('doc');
+const Taboo = require('taboo');
+
+const inputdoc = (req, res) => {
```
Comment (melvinheks) [inline server/checkdoc.js]: rename to saveDoc

File: server/checkdoc.js [hunk 38]
```diff
@@ -0,0 +1,28 @@
...
+  let docinst = new Doc({
+    title: req.body.name,
+    content: req.body.text,
+    owner_id: req.body.owner_id,
+    locked: false,
+    revision_id: 0,
+  });
+
+  let taboo_words_data = Taboo.find({}, function (err, results) {
```
Comment (melvinheks) [inline server/checkdoc.js]: model.find does not return the results. Please read documentation on how to properly use this method

Comment (melvinheks) [inline server/checkdoc.js]: This should not exist here. Perhaps create a separate function that retrieves taboo words and returns a list of pairs of indexes. Rename this file to docHandlers?

File: server/doc.js (hunk 25 above)
Comment (MicLi01) [reply server/doc.js]: It was changed previously already

File: server/taboo.js (hunk 32 above)
Comment (MicLi01) [reply server/taboo.js]: corrected type

File: server/taboo.js (hunk 33 above)
Comment (MicLi01) [reply server/taboo.js]: Corrected type

File: server/taboo.js (hunk 34 above)
Comment (MicLi01) [reply server/taboo.js]: removed

File: server/doc.js (hunk 36 above)
Comment (MicLi01) [reply server/doc.js]: changed to idx_list.push();

File: server/checkdoc.js (hunk 37 above)
Comment (MicLi01) [reply server/checkdoc.js]: changed

File: server/checkdoc.js (hunk 38 above)
Comment (MicLi01) [reply server/checkdoc.js]: changed to send results from response

Comment (MicLi01) [reply server/checkdoc.js]: moved into outside function and renamed file

File: server/docHandlers.js [hunk 39]
```diff
@@ -0,0 +1,31 @@
+const Doc = require('doc');
+const Taboo = require('taboo');
+
+const savedoc = (req, res) => {
+  let docinst = new Doc({
+    title: req.body.name,
+    content: req.body.text,
+    owner_id: req.body.owner_id,
+    locked: false,
+    revision_id: 0,
```
Comment (melvinheks) [inline server/docHandlers.js]: Remove this, this doesn't match schema

File: server/docHandlers.js [hunk 40]
```diff
@@ -0,0 +1,31 @@
...
+  });
+
+  docinst.save(function (err, doc) {
+    if (err) return console.error(err);
+      console.log(doc.name + " saved to Document collection.");
+  });     
+};
+
+const gettaboowords = (req, res) => {
```
Comment (melvinheks) [inline server/docHandlers.js]: This does not make sense to be a function that handles a request and response. Remove this, make it a static method of the taboo schema that returns a list of all taboo words

File: server/doc.js [hunk 41]
```diff
@@ -0,0 +1,36 @@
...
+  }
+})
+
+DocSchema.methods.findAllTabooIdx = function(taboo_words){
+  //A String is considered a word if there is a space
+  //Checks all the words and see if any are in taboo words and replaces those words
+  //returns indexes of all taboo words, format at each index: [start, end + 1]
+  idx_list = [];
+//  for(let i = 0; i < taboo_words.length; i++){
+  for (let taboo_word in taboo_words){
```
Comment (melvinheks) [inline server/doc.js]: Using "in" is incorrect.
This should be (const taboo_word of taboo_words).

File: server/doc.js [hunk 42]
```diff
@@ -0,0 +1,36 @@
...
+    default: Data.now
+  }
+})
+
+DocSchema.methods.findAllTabooIdx = function(taboo_words){
+  //A String is considered a word if there is a space
+  //Checks all the words and see if any are in taboo words and replaces those words
+  //returns indexes of all taboo words, format at each index: [start, end + 1]
+  idx_list = [];
+//  for(let i = 0; i < taboo_words.length; i++){
```
Comment (melvinheks) [inline server/doc.js]: Remove this comment

File: server/doc.js (hunk 35 above)
Comment (melvinheks) [reply server/doc.js]: Address this

File: server/docHandlers.js [hunk 43]
```diff
@@ -0,0 +1,31 @@
+const Doc = require('doc');
+const Taboo = require('taboo');
+
+const savedoc = (req, res) => {
```
Comment (melvinheks) [inline server/docHandlers.js]: Since this handles a request, it should send a response, otherwise the request will timeout

File: server/docHandlers.js (hunk 39 above)
Comment (MicLi01) [reply server/docHandlers.js]: removed

File: server/doc.js (hunk 41 above)
Comment (MicLi01) [reply server/doc.js]: changed

File: server/doc.js (hunk 42 above)
Comment (MicLi01) [reply server/doc.js]: removed

File: server/docHandlers.js (hunk 43 above)
Comment (MicLi01) [reply server/docHandlers.js]: sends message now

File: server/doc.js (hunk 35 above)
Comment (MicLi01) [reply server/doc.js]: moved up

File: server/docHandlers.js [hunk 44]
```diff
@@ -0,0 +1,31 @@
...
+  });
+
+  docinst.save(function (err, doc) {
+    if (err) return console.error(err);
+      console.log(doc.name + " saved to Document collection.");
+  });
+  res.send("Doc saved");     
+};
+
+const gettaboowords = static function() {
```
Comment (melvinheks) [inline server/docHandlers.js]: not correct approach
